general/mem_manager.cpp

```diff
@@ -61,17 +83,424 @@ struct Ledger
 
 static internal::Ledger *maps;
 
+namespace internal
+{
+
+static bool managed;
+
+// *****************************************************************************
+/// The memory space abstract class: one pair (host + device) memory space
+class MemorySpace
```

Comment:
Can we pick a different name? It is confusing to have a class and a structure with the same name

Reply:
I changed that to `Memory::Type`, but I'm open to other suggestions!